src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -245,6 +245,49 @@ void MemoryMngrWithReuse::destroy(void *ptr) {
     dnnl::impl::free(ptr);
 }
 
+void* MemoryMngrRealloc::getRawPtr() const noexcept {
+    return m_data.get();
+}
+
+void MemoryMngrRealloc::setExtBuff(void *ptr, size_t size) {
+    m_useExternalStorage = true;
+    m_memUpperBound = size;
+    m_data = decltype(m_data)(ptr, release);
+}
+
+bool MemoryMngrRealloc::resize(size_t size) {
+    constexpr int cacheLineSize = 64;
+    constexpr size_t growFactor = 2;
+    bool sizeChanged = false;
+    if (size > m_memUpperBound) {
+        size *= growFactor;
```

Comment:
Adding a minimal increment size limitation may be better for very short input and long output?

Reply:
Theoretically yes, but what minimal increment size should we choose? The problem is that it's not that simple to develop optimal allocation strategy for all possible cases. So may be it's better to keep it simple?

Reply:
Sure, modify when truly needed.